solutions/arrays/shuffle-array.js

```diff
@@ -0,0 +1,36 @@
+let fs = require("fs");
+let data = fs.readFileSync(0, 'utf-8');
+let idx = 0;
+data = data.split('\n');
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+
+function shuffle(arr){
+    //Implement this function
+	let leftIndex=0,rightIndex=arr.length/2
+	let answer=[]
+	for(let index=0;index<arr.length/2;index++)

```

Comment:
@DevratAgarwal  instead of using a `for` loop with `index` having no utility, lets write a `while` loop with `letfIndex < arr.length/2` and `rightIndex < arr.length` as conditions. So that the code can be more understandable.